modules/aws/src/smithy4s/query/FormData.scala

```diff
@@ -0,0 +1,55 @@
+package smithy4s.aws.query
+
+final case class FormData(
+    values: Map[List[String], String],
+    primitive: Option[String]
+) {
+  lazy val render: Option[String] =
+    primitive.orElse(
+      if (values.isEmpty) None
+      else
+        Some(
+          values
+            .map { case (segments, value) =>
+              segments.reverse.mkString(".") + "=" + value
+            }
+            .mkString("&")
+        )
+    )
+
+  def addSegment(segment: String): FormData = primitive match {
+    case Some(value) => FormData(Map(List(segment) -> value), None)
+    case None =>
+      FormData(
+        values.map { case (segments, value) =>
+          (segment :: segments, value)
+        },
+        None
+      )
+  }
+
+  def childOf(parent: String): FormData =
```

Comment:
Looking at the [spec](https://smithy.io/2.0/aws/protocols/aws-query-protocol.html?highlight=query%20prot#collections), this doesn't need to exist : the id of the shape is never used in the encoding.

Reply:
Done